### PR TITLE
Run unit and integration tests in Jenkins

### PR DESCRIPTION
**Pull request by eloaverona**
Updates the Jenkinsfile to run rust unit tests and integration tests.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 Cargill Incorporated
```
Comment (vaporos) [inline docker/Dockerfile]: Should we name this Dockerfile something more specific?

Comment (eloaverona) [reply docker/Dockerfile]: Like `tests_dockerfile`?

File: daemon/src/rest_api/routes/mod.rs [hunk 2]
```diff
@@ -82,7 +82,7 @@ mod test {
     use sawtooth_sdk::messaging::stream::{MessageFuture, MessageSender, SendError};
     use std::sync::mpsc::channel;
 
-    static DATABASE_URL: &str = "postgres://grid_test:grid_test@localhost:5433/grid_test";
+    static DATABASE_URL: &str = "postgres://grid_test:grid_test@test_server:5432/grid_test";
```
Comment (agunde406) [inline daemon/src/rest_api/routes/mod.rs]: Typo in commit message. demon -> daemon

File: docker/Dockerfile (hunk 1 above)
Comment (dplumb94) [reply docker/Dockerfile]: I'd suggest `tests` or `run-tests`

File: docker/compose/grid_tests.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+version: "3.6"
```
Comment (agunde406) [inline docker/compose/grid_tests.yaml]: Missing license

File: docker/tests [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+#Build modules with dummy source in order to cache dependencies in Docker image.
+COPY ./contracts/schema/Cargo.toml ./Cargo.toml
+RUN cargo check
+
+WORKDIR /
+RUN USER=root cargo new --bin daemon
+WORKDIR /daemon
+COPY ./daemon/Cargo.toml ./Cargo.toml
+RUN cargo check
```
Comment (rbuysse) [inline docker/tests]: Looks like you're running cargo check as part of building this image. Should this be a separate step in the compose file?

Comment (eloaverona) [reply docker/tests]: I'm running this here so that the rust dependencies can be downloaded and cached in the image.

File: Jenkinsfile [hunk 5]
```diff
@@ -72,8 +72,18 @@ node ('master') {
...
+            sh 'docker-compose -f docker/compose/grid_tests.yaml build --force-rm'
         }
 
+        stage("Run unit tests") {
+            sh 'docker-compose -f docker/compose/grid_tests.yaml up --abort-on-container-exit --exit-code-from grid_tests'
+            sh 'docker-compose -f docker/compose/grid_tests.yaml down'
+        }
+
+         stage("Run integration tests") {
+             sh './bin/run_integration_tests'
```
Comment (rberg2) [inline Jenkinsfile]: There is an extra space on this line

Comment (eloaverona) [reply Jenkinsfile]: Addressed!

File: docker/tests (hunk 4 above)
Comment (eloaverona) [reply docker/tests]: After an offline discussion, I'm marking this conversation as resolved.

**Pull request merged**